test/k8sT/Policies.go

```diff
@@ -281,5 +281,240 @@ var _ = Describe("K8sPolicyTest", func() {
+
+			Expect(returnCode).Should(Equal("200"), "Unable to connect between front and backend:80/")
+
+			By(fmt.Sprintf("running curl %s:80/health from pod %s (shouldn't work)", backendIP, frontendPod))
+
+			returnCode = kubectl.Exec(fmt.Sprintf("kubectl exec -n qa -i %s -- curl --connect-timeout 20 -s -o /dev/null -w \"%%{http_code}\" http://%s:80/health", frontendPod, backendIP)).GetStdOut()
+
+			Expect(returnCode).Should(Equal("403"), fmt.Sprintf("Unexpected connection between frontend and backend; wanted HTTP 403, got: HTTP %s", returnCode))
+
+			/*
+				//TODO -- add this ?
+				kubectl exec -n qa -i ${frontend_pod} -- wrk -t20 -c1000 -d60 "http://${backend_svc_ip}:80/"
+			*/
+		}
+
+		ciliumPodK8s1, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+		Expect(err).Should(BeNil())
+
+		ciliumPodK8s2, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s2)
+		Expect(err).Should(BeNil())
+
+		ciliumPods := []string{ciliumPodK8s1, ciliumPodK8s2}
+
+		// Set debug mode to false for both cilium pods.
+		for _, ciliumPod := range ciliumPods {
+			out, err := kubectl.ExecPodCmd(helpers.KubeSystemNamespace, ciliumPod, "cilium config Debug=false")
+			Expect(err).Should(BeNil(), fmt.Sprintf("error disabling debug mode for cilium pod %s: %s", ciliumPod, out))
+		}
+
+		By("Creating Kubernetes namespace qa")
+		res := kubectl.CreateResource(namespace, qaNs)
+		defer kubectl.DeleteResource(namespace, qaNs)
+		res.ExpectSuccess()
+
+		By("Creating Kubernetes namespace development")
+		res = kubectl.CreateResource(namespace, developmentNs)
+		defer kubectl.DeleteResource(namespace, developmentNs)
+		res.ExpectSuccess()
+
+		resources := []string{"1-frontend.json", "2-backend-server.json", "3-backend.json"}
+		for _, resource := range resources {
+			resourcePath := kubectl.ManifestGet(resource)
+			fmt.Println("resourcePath: %s", resourcePath)
+			res = kubectl.Create(resourcePath)
+			defer kubectl.Delete(resourcePath)
+			res.ExpectSuccess()
+		}
+
+		By("Waiting for endpoints to be ready on k8s-2 node")
+		areEndpointsReady := kubectl.CiliumEndpointWait(ciliumPodK8s2)
+		Expect(areEndpointsReady).Should(BeTrue())
+
+		By("Getting information about pods in qa namespace")
+		res = kubectl.Exec("kubectl get pods -n qa -o wide")
+		log.Infof("%s", res.GetStdOut())
+
+		By("Getting information about pods in development namespace")
+		res = kubectl.Exec("kubectl get pods -n development -o wide")
+		log.Infof("%s", res.GetStdOut())
+
+		By("Getting information about backend service in development namespace")
+		res = kubectl.Exec("kubectl describe svc -n development backend")
+		log.Infof("%s", res.GetStdOut())
+
+		By("Getting K8s services")
+		res = kubectl.Exec("kubectl get svc --all-namespaces")
+		log.Infof("%s", res.GetStdOut())
+
+		pods, err := kubectl.WaitForServiceEndpoints(developmentNs, "", "backend", "80", helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
```

Comment:
Shouldn't this line be swapped with the next one? If there's an error, pods will be false and will we still know which error was it?

Reply:
Fixed.